src/Soil/Soil.jl

```diff
@@ -181,88 +167,62 @@ function dss!(
 end
 
 """
-   StateBC{FT} <: AbstractBC{FT}
+   StateBC{FT} <: AbstractSoilBC{FT}
 
 A simple concrete type of boundary condition, which enforces a
 constant normal state at either the top or bottom of the domain.
 """
-struct StateBC{FT} <: AbstractBC{FT}
+struct StateBC{FT} <: AbstractSoilBC{FT}
     bc::FT
 end
 
 """
-   FluxBC{FT} <: AbstractBC{FT}
+   FluxBC{FT} <: AbstractSoilBC{FT}
 
 A simple concrete type of boundary condition, which enforces a
 constant normal flux at either the top or bottom of the domain.
 """
-struct FluxBC{FT} <: AbstractBC{FT}
+struct FluxBC{FT} <: AbstractSoilBC{FT}
     bc::FT
 end
 
 """
-    TopBoundary{} <: AbstractBoundary{}
-
-A simple object which should be passed into a function to
-indicate that we are considering the top boundary of the soil.
-"""
-struct TopBoundary <: AbstractBoundary end
-
-"""
-    BottomBoundary{} <: AbstractBoundary{}
+    FreeDrainage{FT} <: AbstractSoilBC{FT}
 
-A simple object which should be passed into a function to
-indicate that we are considering the bottom boundary of the soil.
+A concrete type of soil boundary condition, for use at 
+the BottomBoundary only, where the flux is set to be
+`F = -K∇h = -K`.
 """
-struct BottomBoundary <: AbstractBoundary end
+struct FreeDrainage{FT} <: AbstractSoilBC{FT} end
 
 """
-    get_Δz(z)
+    ClimaLSM.boundary_flux(bc::FluxBC, _, Δz, _...)
 
-A function to return a tuple containing the distance between the top boundary
-and its closest center, and the bottom boundary and its closest center, 
-both as Fields.
+A method of boundary fluxes which returns the desired flux.
 """
-function get_Δz(z)
-    # Extract the differences between levels of the face space
-    fs = ClimaLSM.Domains.obtain_face_space(axes(z))
-    z_face = ClimaCore.Fields.coordinate_field(fs).z
-    Δz = ClimaCore.Fields.Δz_field(z_face)
-
-    Δz_top = Fields.level(
-        Δz,
-        ClimaCore.Utilities.PlusHalf(ClimaCore.Spaces.nlevels(fs) - 1),
-    )
-    Δz_bottom = Fields.level(Δz, ClimaCore.Utilities.PlusHalf(0))
-    return Δz_top ./ 2, Δz_bottom ./ 2
-end
-
-"""
-    diffusive_flux(K, x_2, x_1, Δz)
-
-Calculates the diffusive flux of a quantity x (water content, temp, etc).
-Here, x_2 = x(z + Δz) and x_1 = x(z), so x_2 is at a larger z by convention.
-"""
-function diffusive_flux(K, x_2, x_1, Δz)
-    return @. -K * (x_2 - x_1) / Δz
+function ClimaLSM.boundary_flux(
+    bc::FluxBC{FT},
+    boundary::ClimaLSM.AbstractBoundary,
+    Δz,
+    p,
+    _...,
+) where {FT}
+    return bc.bc .+ ClimaCore.Fields.zeros(axes(Δz))
```

Comment:
Maybe add a comment explaining why the boundary condition value is added to this Field object since it may not be obvious from just reading the code